Sources/StreamChatUI/Composer/ChatMenu/EmojiMenu/GifViewController.swift

```diff
@@ -0,0 +1,383 @@
+            self.progressView.isHidden = true
+            if error == nil && self.isSearchActive {
+                self.updateSearchGifView(latestSearchResponse: response, isSearch: isSearch)
+            }
+        })
+    }
+
+    private func updateSearchGifView(latestSearchResponse: GiphyResponse?, isSearch: Bool) {
+        guard let response = latestSearchResponse else { return }
+        isFetchingApiData = false
+        if isSearch {
+            searchGifs.removeAll()
+        }
+        pagingProgressView.isHidden = true
+        bottomView.isHidden = true
+        self.latestSearchResponse = response
+        progressView.isHidden = true
+        collectionView?.alpha = 1.0
+        if searchGifs.isEmpty {
+            searchGifs.append(contentsOf: response.data ?? [])
+            collectionView?.setContentOffset(.zero, animated: false)
+            collectionView?.reloadData()
+        } else {
+            collectionView?.performBatchUpdates({
+                let updateIndex = response.data.enumerated().compactMap { IndexPath(row: $0.offset + self.searchGifs.count, section: 0)} ?? []
+                let indexPath = IndexPath(row: response.data.count ?? 0 , section: 0)
+                searchGifs.append(contentsOf: response.data ?? [])
+                collectionView?.insertItems(at: updateIndex)
+            }, completion: nil)
+        }
+        trendingGifs.removeAll()
+        currentTrendingOffset = 0
+    }
+
+    @objc func btnBackPressed(sender: UIButton!) {
+        dismiss(animated: true, completion: nil)
+    }
+}
+
+@available(iOS 13.0, *)
+extension GifViewController: UISearchBarDelegate {
+    func searchBarShouldBeginEditing(_ searchBar: UISearchBar) -> Bool {
+        if !isSearchEnable {
+            let gifVc = GifViewController(with: true)
+            UIApplication.shared.keyWindow?.rootViewController?.present(gifVc, animated: true, completion: nil)
+        }
+        return isSearchEnable
+    }
+
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        searchBar.resignFirstResponder()
+    }
+
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        searchingText = searchText
+        if searchText.isEmpty {
+            isSearchActive = false
+            getTrendingGifs(isPaginated: false, scrollToTop: true)
+        }
+    }
+}
+
+@available(iOS 13.0, *)
+extension GifViewController: UICollectionViewDelegate {
+
+    func collectionView(_ collectionView: UICollectionView, willDisplay cell: UICollectionViewCell, forItemAt indexPath: IndexPath) {
+        if isFetchingApiData {
+            return
+        }
+        if (isSearchActive ? indexPath.row == searchGifs.count - 1 : indexPath.row == trendingGifs.count - 1) {
+            if isSearchActive {
+                if searchGifs.count < (latestSearchResponse?.pagination.totalCount ?? 0) {
+                    currentSearchOffset = searchGifs.count
```

Comment:
@mdhanif-simformsolutions  Suggestion: Don't put API variables `currentSearchOffset` `currentTrendingOffset` alone. Better to move it in the API call function.

Reply:
Updated this. we don't need it now as we have moved all the logic in the viewmodel itself so managed from there.